pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,7 +398,9 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		node := *res.PrevNode
+		node.ModifiedIndex = res.Node.ModifiedIndex
```

Comment:
So is this * versions?  I can see how this might be a subtle issue in some of the informer changes being made. 

/cc @rrati